OpenMage/magento-lts
# Adds support for "SameSite" cookie property

### Description (*)
This PR adds support to set the SameSite cookie property and is compatible with both the new setcookie signature (>= PHP 7.3) and the old one (< PHP 7.3).

By default the property is set to "None". The value can be changed in the admin-panel under System -> Configuration -> Web -> Session Cookie Management ->  Same-Site

### Related Pull Requests

/

### Fixed Issues (if relevant)

1. Fixes OpenMage/magento-lts#414
2. Fixes https://github.com/mollie/Magento/issues/187

### Manual testing scenarios (*)

1. Open the console i(Network-tab) in your browser and be ready to inspect the Set-cookie header
2. Navigate to your shop to OpenMage shop
2. Confirm that the Set-cookie header contains "SameSite=None"
3. In another tab, go to the admin panel, go to "System -> Configuration -> Web -> Session Cookie Management ->  Same-Site" and update the value to "Strict". (Flush cache...)
4. Go to your initial tab and refresh the page
5. Confirm that the Set-cookie header contains "SameSite=Strict"

### Questions or comments
Open for feedback, first contribution to OpenMage ;)

### Contribution checklist (*)
